doc/tutorial/argument_parser/index.md

```diff
@@ -0,0 +1,414 @@
+
+\snippet test/snippet/argument_parser/validators_3.cpp validator_call
+
+### The seqan3::file_existance_validator
+
+The validator throws a seqan3::parser_invalid_argument exception whenever a given filename (string) does not exist.
+
+\snippet test/snippet/argument_parser/validators_file_existance.cpp validator_call
+
+\assignment{Assignment 8}
+Add a seqan3::file_existance_validator to the first positional option that expects the `file_path`.
+\endassignment
+\solution
+\snippet doc/tutorial/argument_parser/small_snippets.cpp file_existence_validator
+\endsolution
+
+### The seqan3::regex_validator
+
+On construction, the validator receives a pattern for a regular expression.
+The pattern variable will be used for constructing an std::regex and the validator will call std::regex_match on the command line argument.
+
+Note: A regex_match will only return true if the strings matches the pattern completely (in contrast to regex_search which also matches substrings). The struct than acts as a functor that throws a seqan3::parser_invalid_argument exception whenever a given filename (string) is not in the given extension list.
+
+\snippet test/snippet/argument_parser/validators_4.cpp validator_call
+
+## Chaining validators
+
+You can also chain validators using the pipe operator (`|`). The pipe operator is the AND operation for two validators, which means that a value must pass both validators in order to be accepted by the new validator.
+
+For example, you may want a file name that only accepts absolute paths but also must have one out of some given file extensions.
+For this purpose you can chain a seqan3::regex_validator to a seqan3::file_ext_validator:
+
+\snippet test/snippet/argument_parser/validators_chaining.cpp validator_call
+
+You can chain as many validators as you want which will be evaluated one after the other from left to right (first to last).
+
+\assignment{Assignment 9}
+Add a seqan3::file_ext_validator to the first positional option that expects the `file_path` by chaining it to the already present seqan3::file_existance_validator. The file extensions of our application should be restricted to `tsv` since we expect tab separated files.
+\endassignment
+\solution
+\snippet doc/tutorial/argument_parser/solution6.cpp file_existence_validator
+\endsolution
+
+## Create your custom validator
+
+If our validators to not suit your needs, you can always create your own validator. In order to be accepted by the seqan3::argument_parser, your validator must satisfy the following requirements (see also seqan3::validator_concept):
+
+1. It needs to expose a `value_type` type member which identifies the type of variables the validator works on. Currently, the seqan3 validators either have value_type `double` or `std::string`. Since the validator works on every type that has a common reference type to `value_type`, this enables a validator with `value_type = double` to work on all arithmetic values.
```

Comment:
```suggestion
1. It needs to expose a `value_type` type member which identifies the type of variables the validator works on. Currently, the SeqAn3 validators either have value_type `double` or `std::string`. Since the validator works on every type that has a common reference type to `value_type`, it enables a validator with `value_type = double` to work on all arithmetic values.
```